Source/DafnyCore/Auditor/Assumption.cs

```diff
@@ -25,22 +25,35 @@ bool allowedInLibraries
     allowedInLibraries: false);
   public static AssumptionDescription ExternFunction = new(
     issue: "Function with [{:extern}] attribute.",
-    mitigation: "Turn into a [method] with [modifies *] or test thoroughly for lack of side effects.",
-    mitigationIETF: "SHOULD use [method] with [modifies *] instead.",
-    isExplicit: false,
-    allowedInLibraries: false);
-  public static AssumptionDescription ExternWithPrecondition = new(
-    issue: "Declaration with [{:extern}] has a requires clause.",
-    mitigation: "Test any external callers (maybe with [/testContracts]).",
-    mitigationIETF: "MUST test any external callers.",
-    isExplicit: false,
-    allowedInLibraries: false);
-  public static AssumptionDescription ExternWithPostcondition = new(
-    issue: "Declaration with [{:extern}] has a ensures clause.",
-    mitigation: "Test external callee (maybe with [/testContracts]).",
-    mitigationIETF: "MUST test external callee.",
+    mitigation: "Turn into a [method] with [modifies {}] or test thoroughly for lack of side effects.",
+    mitigationIETF: "SHOULD use [method] with [modifies {}] instead.",
     isExplicit: false,
     allowedInLibraries: false);
+
+  public static AssumptionDescription ExternWithPrecondition(bool hasAxiomAttribute) {
```

Comment:
I know I originally suggested this design, but now I'm thinking this should  be treated as such:
* If an extern _does not_ have `{:axiom}`, have the auditor flag any pre- or post-conditions. Don't allow it in libraries.
* If an extern _does_ have `{:axiom}`, have the auditor flag it as an explicit (intentional) assumption, and allow it in libraries. Don't say anything about pre- or post-conditions.
This is more consistent with treatments elsewhere. Having `{:axiom}` on an enclosing scope (e.g., a declaration) silences mention of other assumptions associated with it (e.g., the contract of a declaration).